### PR TITLE
Update chill-protobuf with protobuf 3.17.1

### PR DESCRIPTION
**Pull request by regadas**

### CODE REVIEW

File: chill-protobuf/src/test/protobuf/test_messages.proto [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  int64 target_id = 1;
+  int64 suggested_id = 2;
+  int32 serve_count = 3;
+  repeated int64 recent_clicks = 4;
+  Gender gender = 5;
+}
+
+enum Gender {
+  MALE = 0;
+  FEMALE = 1;
```
Comment (johnynek) [inline chill-protobuf/src/test/protobuf/test_messages.proto:14]: Let's add NONBINARY to this for inclusion.

File: build.sbt [hunk 2]
```diff
@@ -261,10 +261,12 @@ lazy val chillProtobuf = module("protobuf")
...
     autoScalaLibrary := false,
     libraryDependencies ++= Seq(
-      "com.google.protobuf" % "protobuf-java" % "2.3.0" % "provided"
-    )
+      "com.google.protobuf" % "protobuf-java" % "3.17.1" % "provided"
+    ),
+    ProtobufConfig / sourceDirectory := (Test / sourceDirectory).value / "protobuf"
   )
   .dependsOn(chillJava)
+  .enablePlugins(ProtobufPlugin)
```
Comment (johnynek) [inline build.sbt]: I want to be sure this doesn't add scala runtime dependencies to the project. I think this project is Java only currently. And I don't want to wind up pulling in scala library here.

Can we print the Pom and check that the dependencies are as we expect?

**Review by johnynek — CHANGES_REQUESTED** [review]

File: build.sbt (hunk 2 above)
Comment (regadas) [reply build.sbt]: good call out!

```
<?xml version='1.0' encoding='UTF-8'?>
<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0">
    <modelVersion>4.0.0</modelVersion>
    <groupId>com.twitter</groupId>
    <artifactId>chill-protobuf</artifactId>
    <packaging>jar</packaging>
    <description>chill-protobuf</description>
    <version>0.9.5+151-2c4543f6-SNAPSHOT</version>
    <name>chill-protobuf</name>
    <organization>
        <name>com.twitter</name>
    </organization>
    <scm>
        <url>https://github.com/twitter/chill</url>
        <connection>scm:git:https://github.com/twitter/chill.git</connection>
        <developerConnection>scm:git:git@github.com:twitter/chill.git</developerConnection>
    </scm>
    <url>https://github.com/twitter/chill</url>
    <licenses>
        <license>
            <name>Apache 2</name>
            <url>http://www.apache.org/licenses/LICENSE-2.0.txt</url>
            <distribution>repo</distribution>
            <comments>A business-friendly OSS license</comments>
        </license>
    </licenses>
    <developers>
        <developer>
            <id>oscar</id>
            <name>Oscar Boykin</name>
            <url>http://twitter.com/posco</url>
        </developer>
        <developer>
            <id>sritchie</id>
            <name>Sam Ritchie</name>
            <url>http://twitter.com/sritchie</url>
        </developer>
    </developers>
    <dependencies>
        <dependency>
            <groupId>com.twitter</groupId>
            <artifactId>chill-java</artifactId>
            <version>0.9.5+151-2c4543f6-SNAPSHOT</version>
        </dependency>
        <dependency>
            <groupId>org.scalacheck</groupId>
            <artifactId>scalacheck_2.12</artifactId>
            <version>1.15.2</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.scalatest</groupId>
            <artifactId>scalatest_2.12</artifactId>
            <version>3.2.9</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.scalatestplus</groupId>
            <artifactId>scalatestplus-scalacheck_2.12</artifactId>
            <version>3.1.0.0-RC2</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>com.esotericsoftware</groupId>
            <artifactId>kryo-shaded</artifactId>
            <version>4.0.2</version>
        </dependency>
        <dependency>
            <groupId>com.google.protobuf</groupId>
            <artifactId>protobuf-java</artifactId>
            <version>3.17.1</version>
            <scope>provided</scope>
        </dependency>
    </dependencies>
</project>
```

Comment (regadas) [reply build.sbt]: The plugin `sbt-protobuf` does add the `protobuf-java` to `libraryDependencies` as Compile scope which is actually a bit bad. Switched now to https://github.com/thesamet/sbt-protoc which works as expected.

**Review by johnynek — APPROVED** [review]
If the pom looks clean, happy to merge now.

Thanks for taking this.

**Comment by regadas** [comment]
@johnynek Thanks for the review! I'll go ahead and do a release `0.10.0` since we have some breaking changes.

**Pull request merged**